### PR TITLE
 Add setup and deploy erdjs-snippets

### PR DESCRIPTION
**Pull request by CostinCarabas**

### CODE REVIEW

File: erdjs-snippets/lending-pool/lendingPoolInteractor.ts [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+
+    async issueLend(user: ITestUser, tokenIdentifier: string): Promise<ReturnCode> {
+        console.log(`LendingPoolInteractor.issueLend(): address = ${user.address}`);
+
+        // Prepare the interaction
+        let interaction = <Interaction>this.contract.methods
+            .issueLendToken([tokenIdentifier, tokenIdentifier.split("-")[0]])
+            .withGasLimit(120000000)
+            .withValue(50000000000000000)
```
Comment (andreibancioiu) [inline erdjs-snippets/lending-pool/lendingPoolInteractor.ts:152]: You can pass a `TokenPayment.egldFromAmount(0.05)` here.

File: erdjs-snippets/lending-pool/lendingPoolInteractor.ts [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+        return returnCode;
+    }
+
+    async issueBorrow(user: ITestUser, tokenIdentifier: string): Promise<ReturnCode> {
+        console.log(`LendingPoolInteractor.issueBorrow(): address = ${user.address}`);
+
+        // Prepare the interaction
+        let interaction = <Interaction>this.contract.methods
+            .issueBorrowToken([tokenIdentifier, tokenIdentifier.split("-")[0]])
+            .withValue(50000000000000000)
```
Comment (andreibancioiu) [inline erdjs-snippets/lending-pool/lendingPoolInteractor.ts:181]: You can use `TokenPayment.egldFromAmount()` here.

File: erdjs-snippets/lending-pool/lendingPoolInteractor.ts [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+            .withGasLimit(120000000)
```
Comment (andreibancioiu) [inline erdjs-snippets/lending-pool/lendingPoolInteractor.ts:182]: Less gas limit, perhaps? (here, and above / below etc.) Isn't it too much, 120 mil?

File: erdjs-snippets/lending-pool/lendingPoolInteractor.ts [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            .withChainID(this.networkConfig.ChainID);
+
+        // Let's check the interaction, then build the transaction object.
+        let transaction = interaction.check().buildTransaction();
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await user.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
+        let address = SmartContract.computeAddress(transaction.getSender(), transaction.getNonce());
```
Comment (andreibancioiu) [inline erdjs-snippets/lending-pool/lendingPoolInteractor.ts:193]: Remove these two lines from all functions of the interactor, with the exception of the functions that perform the deploy.

File: erdjs-snippets/lending-pool/lendingPoolInteractor.ts [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+        });
+
+        // Set the transaction nonce. The account nonce must be synchronized beforehand.
+        // Also, locally increment the nonce of the deployer (optional).
+        transaction.setNonce(deployer.account.getNonceThenIncrement());
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await deployer.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
```
Comment (andreibancioiu) [inline erdjs-snippets/lending-pool/lendingPoolInteractor.ts:99]: This was the place where `contract.setAddress()` would make sense. You can remove `setAddress()` from functions below. And perhaps add it once, only here.

File: erdjs-snippets/lending-pool/lendingPoolInteractor.ts (hunk 3 above)
Comment (CostinCarabas) [reply erdjs-snippets/lending-pool/lendingPoolInteractor.ts:182]: It's a bit of guessing, I tried with lower gas limit [1], but it wasn't enough. It seems it needs around 90 mil [2]

[1] https://devnet-explorer.elrond.com/transactions/3a5fd1eab0392e3bff6aa2aa90110a9d8fdbbe9756ca4d0d279c2f65aef7269d
[2] https://devnet-explorer.elrond.com/transactions/bcf61a13d55d9c4bde7275eec07df7c20719eaba0c6aa8135a2692779047a35f

File: erdjs-snippets/lending-pool/lendingPoolInteractor.ts [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+        // Prepare the deploy transaction.
+        let transaction = this.contract.deploy({
+            code: code,
+            codeMetadata: new CodeMetadata(),
+            initArguments: [
+                new TokenIdentifierValue(tokenIdentifier),
+                new BigIntValue(0),
+                new BigIntValue(40000000),
+                new BigIntValue(1000000000),
+                new BigIntValue(800000000),
```
Comment (CostinCarabas) [inline erdjs-snippets/lending-pool/lendingPoolInteractor.ts:48]: @andreibancioiu , I don't like magic numbers in the code. Is there a nice way of aggregating them into an import file? Do you have an example?

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**